mongodbforms/fields.py

```diff
@@ -76,17 +76,18 @@ class ReferenceField(forms.ChoiceField):
     Reference field for mongo forms. Inspired by
     `django.forms.models.ModelChoiceField`.
     """
-    def __init__(self, queryset, empty_label="---------", *args, **kwargs):
+    def __init__(self, document, empty_label="---------", *args, **kwargs):
         forms.Field.__init__(self, *args, **kwargs)
         self.empty_label = empty_label
-        self.queryset = queryset
+        self._document = document
 
     def _get_queryset(self):
-        return self._queryset.clone()
+        return self._document.objects.clone()
 
     def _set_queryset(self, queryset):
-        self._queryset = queryset
+        # COMPAT: queryset always based on _document, but this setter is used for deepcopy()
         self.widget.choices = self.choices
```

Comment:
This side effect is really weird. Any way you can comment to explain why it's done this way?

Reply:
No idea. I assume it's to reload the possible choices whenever the queryset is changed, which technically shouldn't be possible past the `__init__` with these changes.
I don't see any obvious calls to this setter in the rest of the code, so I'll move that stuff to the `__init__`.

Reply:
4de9ab5 :white_check_mark: